modular_skyrat/modules/reagent_forging/code/crafting_bench.dm

```diff
@@ -0,0 +1,242 @@
+		"Horseshoes" = /obj/item/clothing/shoes/horseshoe,
+		"Ring" = /obj/item/clothing/gloves/ring/reagent_clothing,
+		"Collar" = /obj/item/clothing/neck/kink_collar/reagent_clothing,
+		"Handcuffs" = /obj/item/restraints/handcuffs/reagent_clothing,
+		"Borer Cage" = /obj/item/cortical_cage,
+		"Pavise Shield" = /obj/item/shield/riot/buckler/reagent_weapon/pavise,
+		"Buckler Shield" = /obj/item/shield/riot/buckler/reagent_weapon,
+		"Coil" = /obj/item/forging/coil,
+	)
+
+/obj/structure/reagent_crafting_bench/examine(mob/user)
+	. = ..()
+	if(goal_name)
+		. += span_notice("Goal Item: [goal_name]")
+		. += span_notice("When you have the necessary materials, begin hammering!<br>")
+		if(required_chain)
+			. += span_warning("[chain_amount]/[required_chain] chains stored.")
+		if(required_plate)
+			. += span_warning("[plate_amount]/[required_plate] plates stored.")
+		if(required_coil)
+			. += span_warning("[coil_amount]/[required_coil] coils stored.")
+	if(length(contents))
+		. += span_notice("Held Item: [contents[1]]")
+
+/obj/structure/reagent_crafting_bench/update_appearance(updates)
+	. = ..()
+	icon_state = "crafting_bench_[length(contents) ? "full" : "empty"]"
+
+//when picking a design or clearing a design
+/obj/structure/reagent_crafting_bench/attack_hand(mob/living/user, list/modifiers)
+	. = ..()
+	if(length(contents))
+		var/obj/item/moving_item = contents[1]
+		user.put_in_hands(moving_item)
+		balloon_alert(user, "item retrieved!")
+		update_appearance()
+		return
+	if(goal_item_path)
+		clear_required()
+		balloon_alert_to_viewers("table cleared!")
+		update_appearance()
+		return
+	var/target_choice = tgui_input_list(user, "Which item would you like to craft?", "Crafting Choice", allowed_choices)
+	if(!target_choice)
+		balloon_alert(user, "no choice made!")
+		return
+	goal_name = target_choice
+	goal_item_path = allowed_choices[target_choice]
+	switch(target_choice)
+		if("Chain Helmet")
+			required_chain = 5
+		if("Chain Armor")
+			required_chain = 6
+		if("Chain Gloves")
+			required_chain = 4
+		if("Chain Boots")
+			required_chain = 4
+		if("Plated Boots")
+			required_plate = 4
+		if("Horseshoes")
+			required_chain = 4
+		if("Ring")
+			required_chain = 2
+		if("Collar")
+			required_chain = 3
+		if("Handcuffs")
+			required_chain = 10
+		if("Borer Cage")
+			required_plate = 6
+		if("Pavise Shield")
+			required_plate = 8
+		if("Buckler Shield")
+			required_plate = 5
+		if("Coil")
+			required_chain = 2
+	if(!required_hits)
+		required_hits = (required_chain * 2) + (required_plate * 2) + (required_coil * 2)
+	balloon_alert(user, "choice made!")
+	update_appearance()
+
+/obj/structure/reagent_crafting_bench/proc/clear_required()
+	required_hits = 0
+	current_hits = 0
+	goal_item_path = null
+	goal_name = null
+	required_chain = 0
+	required_plate = 0
+	required_coil = 0
+
+/obj/structure/reagent_crafting_bench/proc/check_required_materials()
+	if(chain_amount < required_chain)
+		return FALSE
```

Comment:
this might be a good place to have a to_chat or a balloon_alert saying "not enough [material]!"

applies to all the other if()s in this proc

Reply:
only one instance of the proc used, where there is a balloon alert there.